docs/rfcs/0001-integration-with-dune.md

```diff
@@ -0,0 +1,169 @@
+- Start Date: 2021-05-31
+- RFC PR: https://github.com/melange-re/melange/pull/200
+
+# Summary
+
+To allow Melange users to leverage Dune `library` and `executable` stanzas as
+it is currently possible with any other OCaml project.
+
+# Basic example
+
+Dune would allow to use Melange as another cross-compiler, e.g.
+
+```dune
+(context (default (targets (melange))))
+```
+
+Then, users would be able to define `library` and `executable` stanzas that
+would be built by calling Melange binary `ocamlc`. To define the output format
+and extension, a new stanza `melange` would be supported in `dune` files:
+```dune
+(executable
+    (name foo)
+    (modes melange)
+    (melange (module es6) (suffix bs.js))
```

Comment:
Really nice

Reply:
I'm not sure tbh 😅 ideally i'd like to couple module and suffix together, with a kind of list of tuples like `(es6, bs.js)` but idk how to express this in dune s-expr language.

Reply:
I think we should require an executable per target, the executable may be just a noop wrapper, all the `cmj` will already be shared between each target anyway.

```lisp
(executable (name app_esm) (melange (module esm) (suffix js)) (libraries app))
(executable (name app_cjs) (melange (module cjs) (suffix bs.js)) (libraries app))
```